orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/PipelineStartTracker.groovy

```diff
@@ -48,18 +49,33 @@ class PipelineStartTracker {
     pipelineStack.add(PIPELINE_STARTED_ALL, executionId)
   }
 
-  boolean queueIfNotStarted(String pipelineConfigId, String executionId) {
-    def allRunningExecutionIds = executionRepository.retrievePipelinesForPipelineConfigId(
-      pipelineConfigId,
-      new ExecutionRepository.ExecutionCriteria(limit: Integer.MAX_VALUE, statuses: [ExecutionStatus.RUNNING.toString()])
-    ).subscribeOn(Schedulers.io()).toList().toBlocking().single()*.id
-
-    getStartedPipelines(pipelineConfigId).each { String eId ->
-      if (!allRunningExecutionIds.contains(eId)) {
-        log.info("No running execution found for `${pipelineConfigId}:${eId}`, marking as finished")
-        markAsFinished(pipelineConfigId, eId)
+  /**
+   * Checks all waiting pipelines to see if the pipeline they're waiting for has
+   * finished and if so starts them up.
+   */
+  void triggerQueuedPipelines() {
+    allStartedExecutions.each { startedExecutionId ->
+      try {
+        def execution = executionRepository.retrievePipeline(startedExecutionId)
+        if (execution.status.complete) {
+          processPipelines(execution)
+        }
+      } catch (ExecutionNotFoundException ignored) {
+        log.warn("Unable to update pipeline status for missing execution (executionId: ${startedExecutionId})")
+      } catch (Exception e) {
+        log.error('failed to update pipeline status', e)
```

Comment:
I know this was just copied from elsewhere ... but it should include the executionId in the logged message.